src/shared/lib/StoreUtils.ts

```diff
@@ -465,6 +465,30 @@ export async function fetchCnaOncoKbData(uniqueSampleKeyToTumorType:{[uniqueSamp
     }
 }
 
+export async function fetchCnaOncoKbDataWithGeneMolecularData(uniqueSampleKeyToTumorType:{[uniqueSampleKey: string]: string},
+                                         annotatedGenes:{[entrezGeneId:number]:boolean},
+                                         geneMolecularData:MobxPromise<GeneMolecularData[]>,
+                                          molecularProfileIdToMolecularProfile:{[molecularProfileId:string]:MolecularProfile},
+                                         client: OncoKbAPI = oncokbClient)
+{
+    if (!geneMolecularData.result || geneMolecularData.result.length === 0) {
+        return ONCOKB_DEFAULT;
+    }
+    else
+    {
+        const alterationsToQuery = _.filter(geneMolecularData.result, d=>{
+            return molecularProfileIdToMolecularProfile[d.molecularProfileId].molecularAlterationType === "COPY_NUMBER_ALTERATION" &&
+                !!annotatedGenes[d.entrezGeneId];
+        });
+        const queryVariants = _.uniqBy(_.map(alterationsToQuery, (datum: GeneMolecularData) => {
+            return generateQueryVariant(datum.entrezGeneId,
+                cancerTypeForOncoKb(datum.uniqueSampleKey, uniqueSampleKeyToTumorType),
+                getAlterationString(parseInt(datum.value, 10)));
+        }), "id");
```

Comment:
as nice as it is to use the string shortcut, we should use (molduclarData)=>molecularData.id  
to get the typescript safety.   We should just establish that habit.

Reply:
I just copied this over but you're right